packages/astro/src/core/routing/request.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * Utility function that creates a new `Request` with a new URL from an old `Request`.
+ * 
+ * @param newUrl The new `URL`
+ * @param oldRequest The old `Request`
+ */
+export async function copyRequest(newUrl: URL, oldRequest: Request): Promise<Request> {
+	const body = oldRequest.headers.get('content-type') ? oldRequest.blob() : Promise.resolve(undefined);
+	return body.then((requestBody) => {
+		return new Request(newUrl, {
+			method: oldRequest.method,
+			headers: oldRequest.headers,
+			body: requestBody,
```

Comment:
I was wondering why we need to copy all these manually, and it seems like the spec never quite handled `body`. It also seems similar to https://stackoverflow.com/a/34641566 where there's a comment that we should only copy the body if it's not GET or HEAD instead? Or do you have a reason to base on content-type?

Also, I wonder if `body: oldRequest.body` would be enough, since the body is already a valid type for the input 🤔

Reply:
It would be if it has not been ready.

Reply:
The Stackoverflow comment was old and not up to date with the specs. I found a way to copy the body without incurring into errors: [`e6b975e` (#11182)](https://github.com/withastro/astro/pull/11182/commits/e6b975ef208d11bb054f04f0bc599b717100f222)